model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -57,26 +57,20 @@ def nodes_to_remove(self, graph: Graph, match: dict):
 
     def replace_sub_graph(self, graph: Graph, match: dict):
         decoder_node = match['decoder']
+        decoder_node_name = decoder_node.soft_get('name')
         graph.remove_edge(decoder_node.id, match['sparse_to_dense'].id)
         graph.remove_edge(decoder_node.id, match['cast'].id)
         match['sparse_to_dense'].replace_node(decoder_node)
 
-        # update the TensorFlow infer function for the CTCGreedyDecoder to make necessary changes with the second input
-        decoder_node['old_infer'] = decoder_node.infer
-        decoder_node.infer = __class__.tf_greedy_decoder_infer
-        return {}
-
-    @staticmethod
-    def tf_greedy_decoder_infer(node: Node):
-        sequence_length_node = node.in_node(1)
+        sequence_length_node = decoder_node.in_node(1)
         if sequence_length_node.value is None:
             raise Error('The second input to the CTCGreedyDecoder node "{}" is not constant. This case is not '
-                        'supported with the Inference Engine.'.format(node.soft_get('name')))
-        # the batch size is the dimension with index 1 for the layer CTCGreedyDecoder
-        new_value = np.ones([node.in_node(0).shape[1], sequence_length_node.value[0]])
-        new_value[:, 0] = 0
-        new_value = np.transpose(new_value)
-        sequence_length_node.value = new_value
-        sequence_length_node.shape = int64_array(sequence_length_node.value.shape)
+                        'supported with the Inference Engine.'.format(decoder_node_name))
+        mask_value = np.ones([decoder_node.in_node(0).shape[1], sequence_length_node.value[0]])
```

Comment:
Looks like this transformation are not reshape-able. Maybe we can implement sub-graph like in PR https://github.com/openvinotoolkit/openvino/pull/768?

Reply:
Anton, absolutely agree, the transformation generates a model non-reshapable along batch dimension. Any ideas did not come to my mind how to overcome it. I will take a look at you PR and cherry-pick necessary changes. Thanks

Reply:
We agreed to do refactoring and fixing #31610 in the scope of other commit. Now I just refactored to have common infer function for CTCGreedyDecoder operation.
The other transformation CTCGreedyDecoder+CTCLoss preserves reshape-ability.